### ISSUE TITLE
expected comma after object element

### ISSUE THREAD

**Issue by aexvir** [goccy/go-json#340]
similarly as reported in https://github.com/goccy/go-json/issues/217, we're hitting an issue where `go-json` randomly fails parsing payloads, all of which are completely valid json, correctly parseable by `encoding/json` and `jsoniter`

and it's not clear to me what's causing this...

attaching the debugger and checking the code, I'd say that it's something related to the cursor being incremented incorrectly when going over some specific character combination, as just modifying the json file by removing whitespaces, or removing a couple random elements does make the parsing successful

you can reproduce the issue with the following code:
```go
package main

import (
	"bytes"
	"fmt"
	"os"
	"strings"

	"github.com/goccy/go-json"
)

type Payload struct {
	Data []struct {
		Src string `json:"src"`
		Dst string `json:"dst"`
	} `json:"data"`
}

func main() {
	local, _ := os.ReadFile("dump.json")

	var pld Payload

	err := json.NewDecoder(bytes.NewReader(local)).Decode(&pld)
	if err != nil {
		fmt.Printf("failed on regular file: %s\n", err)

		// random change that makes it pass
		// other changes like remoing some items also makes it pass
		min := strings.ReplaceAll(string(local), " ", "")
		err = json.NewDecoder(strings.NewReader(min)).Decode(&pld)
		if err != nil {
			fmt.Printf("failed on minified file: %s\n", err)
		}
	}

	fmt.Printf("\nloaded %d entries", len(pld.Data))
}
```

with the following `dump.json` file: https://pastebin.com/pvqJWmn9
apologies for not being able to give a smaller example

---

if it helps to get you started, conditionally breaking in `struct.go:673` when `s.offset == 32150` will leave you on the last three iterations before failing, as that's trying to parse the chunk
```
"max_duration": 1295}, {"src": "KBR", "dst": "LGK", "stopovers": 0�
```
which ends up failing when checking if that last `0` is a comma

**Comment by aexvir** [comment]
@orisano I've checked your patch and unfortunately it still fails to parse this json, with the same error 😕 
I've used the exact same code and payload I've added to this issue report, and then
```bash
go mod edit -replace github.com/goccy/go-json=github.com/orisano/go-json@81519c48d8dd664cd52ccd5e77ec2a7cc1580e09
```
so that fix doesn't seem to be covering this use case unfortunately

**Comment by orisano** [comment]
Sorry, I misunderstood this issue.
Could you confirm this PR? https://github.com/goccy/go-json/pull/341

**Comment by aexvir** [comment]
damn you're fast 😅
I can confirm that fix correctly addresses this problem
thanks!

btw... I'm thinking about making a branch in my own fork including the three fixes you've opened pull requests for
mainly to continue testing `go-json` as replacement for jsoniter on our current codebase

do you expect any issue with this?

**Comment by aexvir** [comment]
oh... nevermind, it seems that the fix for this issue has been merged to master just right now 🤣 
so I guess the others might also get merged? will keep an eye on this

**Comment by aexvir** [comment]
all merged, amazing 🎉 
thanks @goccy !🙂

**Comment by goccy** [comment]
I just released this fixing as v0.9.5 :)
